frameworks/MLNet/exec.py

```diff
@@ -0,0 +1,96 @@
+import logging
+import math
+import os
+import pandas as pd
+from amlb.benchmark import TaskConfig
+from amlb.data import Dataset
+from amlb.datautils import reorder_dataset
+from amlb.results import NoResultError, save_predictions
+from amlb.utils import dir_of, path_from_split, run_cmd, split_path, Timer
+import json
+from frameworks.shared.callee import call_run, result, save_metadata,  output_subdir
+
+log = logging.getLogger(__name__)
+
+def run(dataset: Dataset, config: TaskConfig):
+    log.info(f"\n**** MLNet [v{config.framework_version}] ****\n")
+    save_metadata(config)
+
+    avaible_task_list = ['classification', 'regression']
+    if config.type not in avaible_task_list:
+        raise ValueError(f'{config.type} is not supported.')
+
+    dir_path = os.path.dirname(os.path.realpath(__file__))
+    DOTNET_INSTALL_DIR = os.path.join(dir_path, 'lib')
+    os.environ['MODELBUILDER_AUTOML'] = 'NNI'
+    os.environ['DOTNET_ROOT'] = DOTNET_INSTALL_DIR
+    os.environ['MLNetCLIEnablePredict'] = 'True'
+    mlnet = os.path.join(DOTNET_INSTALL_DIR, 'mlnet')
+    temp_output_folder = output_subdir('', config=config)
```

Comment:
there are lot of things going into this temp folder, and as it is inside the output dir, all the files created there get uploaded to s3 at the end of the benchmark when it runs on aws, and at the end, downloaded to the machine that started the benchmark, so we want to avoid downloading unnecessary artifacts.

I would recommend to create this tmp folder using:
```
with tempfile.TemporaryDirectory() as tmpdir:
     ...
 ```
 to ensure that the content gets deleted at the end.
 
 if you want to keep the logs, or various artifacts, we accept a param named `_save_artifacts` (see `autosklearn` for an example), for example a user creating its custom framework:
 ```
 myMLNet:
    extends: MLNet
    params:
       _save_artifact = ['logs', 'models']
```
would be able to retrieve the framework logs and the serialized models.
In your case it would mean sth like:
```python
artifacts = config.framework_params.get('_save_artifacts', [])
log_dir = output_subdir('logs', config=config) if 'logs' in artifacts else os.path.join(tmpdir, 'logs')
mlnet_output_dir = output_subdir('models', config=config) if 'models' in artifacts else os.path.join(tmpdir, 'models')
```
this way the `config.output_dir` is not polluted if the extra artifacts were not requested by the user.

Reply:
Sure

Reply:
### Resolved